### PR TITLE
Chore: Migrate to appcontext from object store auth

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

`SignedInUser` can now be pulled from context using `appcontext.WithUser`

**Why do we need this feature?**

This 

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/store/object/dummy/dummy_server.go [hunk 1]
```diff
@@ -378,7 +385,7 @@ func (i *dummyObjectServer) Search(ctx context.Context, r *object.ObjectSearchRe
 // This sets the TenantId on the request GRN
 func getFullGRN(ctx context.Context, grn *object.GRN) *object.GRN {
 	if grn.TenantId == 0 {
-		modifier := store.UserFromContext(ctx)
+		modifier := appcontext.MustUser(ctx)
```
Comment (ryantxu) [inline pkg/services/store/object/dummy/dummy_server.go:388]: I get nervous when I see this... panic!  but I i guess that is what happens here already

**Review by ryantxu — APPROVED** [review]
this is a good step forward, I do think we want a more general `Identity` interface/api in the SDK.  I started:

https://github.com/grafana/grafana-plugin-sdk-go/compare/auth-interface

as an attempt to see how it would look

**Comment by toddtreece** [comment]
@ryantxu should we just skip this one and jump to that? it would be good to avoid using signedInUser directly

**Comment by toddtreece** [comment]
closing this in favor of https://github.com/grafana/grafana-plugin-sdk-go/compare/auth-interface

**Pull request closed without merging**